pkg/adaptor/mongodb/reader.go

```diff
@@ -95,7 +96,10 @@ func (r *Reader) iterateCollection(mgoSession *mgo.Session, in <-chan string, do
 		defer close(out)
 		for {
 			select {
-			case c := <-in:
+			case c, ok := <-in:
+				if !ok {
+					return
+				}
 				log.With("collection", c).Infoln("iterating...")
 				canReissueQuery := r.requeryable(c, mgoSession)
 				var lastID interface{}
```

Comment:
good catch, looks like setting `lastID` got lost in the refactor. I'll get another PR together for that.